bittide/src/Bittide/ProcessingElement/Util.hs

```diff
@@ -22,25 +21,43 @@ import qualified Data.ByteString as BS
 import qualified Data.IntMap as I
 import qualified Data.List as L
 
+-- | Given a `Maybe Int` and a list, if the `Maybe Int` is `Just s`, pad the list to
+-- size `s` with the given element. If the length of the list is greater than `s`,
+-- throw an error. If the `Maybe Int` is `Nothing`, return the list as is.
+padToSize :: Maybe Int -> a -> [a] -> [a]
+padToSize Nothing _ l = l
+padToSize (Just s) a xs
+  | s >= l = xs <> L.replicate (s - l) a
+  | otherwise = error "Bittide.ProcessingElement.Util: List is longer than the specified size"
+  where
+    l = L.length xs
+
+
 -- | Given the path to an elf file, the path to a device tree and a starting address
 --  for the device tree. Return a 3 tuple containing:
 --  (initial program counter, instruction memory blob, data memory blob)
 memBlobsFromElf ::
   -- | How the words should be ordered in the memBlob
   ByteOrder ->
+  -- | Optional size to which we should pad the instruction memBlob and data memBlob
+  (Maybe Int, Maybe Int) ->
   -- | Source file, assumed to be Little Endian.
   FilePath ->
   -- | Optional tuple of starting address and filepath to a device tree.
   Maybe (I.Key, FilePath) ->
   -- | (instruction memBlob, data memBlob)
    Q Exp
-memBlobsFromElf byteOrder elfPath maybeDeviceTree = do
+memBlobsFromElf byteOrder (iSize, dSize) elfPath maybeDeviceTree = do
```

Comment:
I think you should error if the given `iSize` > detected `iSize`. Same for `dSize`.

Reply:
Already happens in `padToSize`